src/Serializers/SlottySerializer.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Wikibase\DataModel\Serializers;
+
+use Serializers\DispatchableSerializer;
+use Serializers\Exceptions\SerializationException;
+use Wikibase\DataModel\Slotty;
+
+/**
+ * Package private
+ *
+ * @licence GNU GPL v2+
+ * @author Adam Shorland
+ * @since 1.9.0
+ */
+class SlottySerializer implements DispatchableSerializer {
+
+	/**
+	 * @since 3.0
+	 *
+	 * @param mixed $object
+	 *
+	 * @return boolean
+	 */
+	public function isSerializerFor( $object ) {
+		return $object instanceof Slotty;
+	}
+
+	/**
+	 * @since 1.0
+	 *
+	 * @param Slotty $object
+	 *
+	 * @return array|int|string|bool|float A possibly nested structure consisting of only arrays and scalar values
+	 * @throws SerializationException
+	 */
+	public function serialize( $object ) {
+		return $object->getSlots();
```

Comment:
I do have two major problems with this proposal:

1. The name is [killing me](http://dict.leo.org/?search=slutty).
2. What's the point of having support for "slots" that can contain nothing but scalar values? What this basically means is: All code working with these "slots" must do it's own (de)serialization to make sure getSlots only returns values that are already in the correct intermediate array format for the JSON and XML serialization. The knowledge about how to (de)serialize a specific "slot" will be spread across different components. If you do this the other way around and have one or many actual "SlotSerializers" that know how to (de)serialize all supported "slots", what's the difference and advantage compared to an actual change to the DataModel and (de)serializers then?

TL;DR: The effect of this proposal is that (de)serializer code will not be in a (de)serializer component any more, but copy-pasted all over the code base where one of the fields must be (de)serialized.